### ISSUE TITLE
Hosting FEA tool on Compute

### ISSUE THREAD

**Issue by Marco-Pellegrino** [mcneel/compute.rhino3d.appserver#78]
Hello.

I am trying to explore the use of a web app to solve some simple FEA Grasshopper model but I am having some issue.
The tool required to run an external application (I do it with `subprocess`) from which we can read an output file.
Is it possible?
I am getting this error message but I am not able to get any tip.

![image](https://user-images.githubusercontent.com/53620289/134698505-d73b86a3-72a6-4b72-ac3b-28249f0b809d.png)

**Comment by sbaer** [comment]
What does you python script look like?

**Comment by Marco-Pellegrino** [comment]
Hi Steve!

Thanks for jumping in.

I am posting here the Grasshopper script in case you want to have a better look and try your self.
I am going to do a bit of debugging my self deleleting some component until I do understand where the issue is happening.

It requires to download Alpaca4d tool (which I have attached to this thread). Remember to unblock the file and to install the tool inside a Folder called "Alpaca4d".

I have tried to summarise what the tool is trying to do in the picture below.
I hope it makes sense.

![image](https://user-images.githubusercontent.com/53620289/135112666-d0edf02f-d68a-48bd-8d83-7e876714a477.png)


[FabianEidnerV
[Alpaca4d.zip](https://github.com/mcneel/compute.rhino3d.appserver/files/7245238/Alpaca4d.zip)
ase.zip](https://github.com/mcneel/compute.rhino3d.appserver/files/7245161/FabianEidnerVase.zip)

**Comment by Marco-Pellegrino** [comment]
The error seems to happen in the "assemble" component.
The latest output that I can get without any error is when I remove that part.

You can look inside the code of the assemble if you want.

![image](https://user-images.githubusercontent.com/53620289/135656012-1c469e63-e507-43d4-a59b-63bd0f1479a9.png)

**Comment by sbaer** [comment]
You can use most of RhinoCommon and ghpythonlib. The area you should stay away from are RhinoDoc and it's views. There is typically no RhinoDoc when running headless

**Comment by Marco-Pellegrino** [comment]
thanks @sbaer !!!
The RhinoDoc makes me wonder where it could be the issue and it is finally working!!!!!

The issue was in rhinoscriptsintax. I was using rs.DocumentPath() to find the working folder but I have solved it using os.getcwd()

I am so glad we have managed to solve it! We can finally run some FEA on the web.

![image](https://user-images.githubusercontent.com/53620289/139561019-d1826dff-e82f-4ac5-9f30-3070d4f7c99d.png)
